### PR TITLE
Modifying vanilla prices

### PR DESCRIPTION
**Pull request by esztermarton**
editing vanilla prices to take in cost_of_carries and riskless_rate while keeping compatibility with version that takes in discount_factors.

### CODE REVIEW

File: tf_quant_finance/black_scholes/vanilla_prices_test.py [hunk 1]
```diff
@@ -134,6 +134,34 @@ def test_price_vol_and_expiry_scaling(self):
...
+    spots = np.array([80.0, 90.0, 100.0, 110.0, 120.0] * 2)
+    strikes = np.array([100.0] * 10)
+    risk_free_rates = 0.08
+    volatilities = 0.2
+    expiries = 0.25
+
+    is_call_options = np.array([True] * 5 + [False] * 5)
+    cost_of_carries = -0.04
+    computed_prices = self.evaluate(
+      tff.black_scholes.option_price(
```
Comment (cyrilchim) [inline tf_quant_finance/black_scholes/vanilla_prices_test.py:149]: This function has a lot of args. We decided to go by keywords for such functions and by positions. Could you please add volatilities=volatilities, strikes=strikes, expiries=expriries

File: tf_quant_finance/black_scholes/vanilla_prices.py [hunk 2]
```diff
@@ -97,21 +118,35 @@ def option_price(volatilities,
...
+      discount_factors = tf.exp(-risk_free_rates * expiries)
 
     if forwards is not None:
       forwards = tf.convert_to_tensor(forwards, dtype=dtype, name='forwards')
     else:
       spots = tf.convert_to_tensor(spots, dtype=dtype, name='spots')
-      forwards = spots / discount_factors
+      forwards = spots * tf.exp(cost_of_carries * expiries)
 
     sqrt_var = volatilities * tf.math.sqrt(expiries)
```
Comment (cyrilchim) [inline tf_quant_finance/black_scholes/vanilla_prices.py]: Please rename `sqrt_var` to `total_var`. I think the bug is here:

    total_vol = volatilities * tf.math.sqrt(expiries)
    d1 = (tf.math.log(forwards / strikes) + ( total_vol * total_vol / 2)) / total_vol
    d2 = d1 - total_vol

You don't need `+ cost_of_carries * expiries` (see [this](https://en.wikipedia.org/wiki/Black%E2%80%93Scholes_model#Alternative_formulation) formula on wiki)

with this change
prices = option_price(volatilities,
    volatilities,
    strikes,
    expiries,
    spots=spots,
    risk_free_rates=risk_free_rates,
    cost_of_carries=cost_of_carries,
    is_call_options=is_call_options,
    dtype=tf.float32)

np.round(prices.numpy(), 2)
gives 
array([ 0.03,  0.57,  3.42,  9.85, 18.62, 20.41, 11.25,  4.4 ,  1.12,
        0.18], dtype=float32)

which is the same as in the paper

File: tf_quant_finance/black_scholes/vanilla_prices.py [hunk 3]
```diff
@@ -21,6 +22,8 @@ def option_price(volatilities,
                  expiries,
                  spots=None,
                  forwards=None,
+                 risk_free_rates=None,
```
Comment (cyrilchim) [inline tf_quant_finance/black_scholes/vanilla_prices.py]: We have some other contribution pending and would like to keep naming consistent. Please rename `risk_free_rates` to `discount_rates`

**Review by cyrilchim — COMMENTED** [review]
Please resolve the other comments from the precious pull requests.
After adding the dividends, this can go to TFF straightaway.
Thank you very much! This is a lot effort and we really appreciate it. Well done!

File: tf_quant_finance/black_scholes/vanilla_prices.py (hunk 2 above)
Comment (esztermarton) [reply tf_quant_finance/black_scholes/vanilla_prices.py]: You are so right. I just redid the derivation - I had misplaced an S for an F in the following formula from [the paper](https://deriscope.com/docs/Barone_Adesi_Whaley_1987.pdf) :'(
Screenshot 2020-02-25 at 22 06 04

**Pull request closed without merging**